### PR TITLE
Add test && Define type for keyboard

### PR DESCRIPTION
**Pull request by chiehw**
1. [x11: refactor key down & up](https://github.com/rustdesk/rustdesk/commit/1c03ad34abd7397a95fbc4716be983648056015c): to get error message.
2. [refactor grab loop](https://github.com/rustdesk/rustdesk/commit/9aaac0e3ee316d1eb89b733d62fe67ecee01f29e): to reduce duplication of code. 
3. [Add keyboard test](https://github.com/rustdesk/rustdesk/commit/d39582d9d48bcc68a9bb95e18631875f55d6d132): to simplify test.
4. [Define new type for keyboard](https://github.com/rustdesk/rustdesk/commit/4c400b2660c06c581285512a190001b36499d20f): to support Translate mode.
5. [Add RawKeyEvent to proto](https://github.com/rustdesk/rustdesk/commit/265acb56697ee242eb6b99ebced47a0baebea676): to send more info on keystrokes.

### CODE REVIEW

**Comment by fufesou** [comment]
`libxkbcommon-x11-dev` is required on Linux, while it is not added to dependencies on the workflow.
Is `libxkbcommon-x11-dev` neccessary?

**Comment by fufesou** [comment]
"Translate mode", "Linux(Fr) -> Win(US)"
All translate chars are incorrect.

**Comment by chiehw** [comment]
> libxkbcommon-x11-dev

xkbcommon is a library for handling of keyboard descriptions, including loading them from disk, parsing them and handling their state. It's mainly meant for client toolkits, window systems, and other system applications;

Currently that includes:

- [Wayland](https://wayland.freedesktop.org/)
- [kmscon](https://github.com/dvdhrm/kmscon/wiki/KMSCON)
- [GTK](https://www.gtk.org/)
- [Qt](https://www.qt.io/)
- [Clutter](https://live.gnome.org/Clutter)
- [KWin](https://userbase.kde.org/KWin)

---

xkbcommon supports more than x11, it can better handle the simulation on the Linux side.

File: src/ui_session_interface.rs [hunk 1]
```diff
@@ -1011,21 +1014,25 @@ impl<T: InvokeUiSession> Interface for Session<T> {
...
-                let ck = match ck {
-                    ControlKey::Control => ControlKey::Meta,
-                    ControlKey::Meta => ControlKey::Control,
-                    ControlKey::RControl => ControlKey::Meta,
-                    ControlKey::RWin => ControlKey::Control,
-                    _ => ck,
-                };
-                hbb_common::protobuf::EnumOrUnknown::new(ck)
-            }).collect();
+        if allow_swap_key {
```
Comment (rustdesk) [inline src/ui_session_interface.rs]: duplicate with line 391

**Review by rustdesk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**